django/db/models/sql/compiler.py

```diff
@@ -1626,7 +1626,7 @@ def pre_sql_setup(self):
 
         # Now we adjust the current query: reset the where clause and get rid
         # of all the tables we don't need (since they're in the sub-select).
-        self.query.where = self.query.where_class()
+        self.query.where = WhereNode()
```

Comment:
Perhaps this could be encapsulated with a `Query.clear_where()` method (could be called in `subqueries.py` too).